client/connection.go

```diff
@@ -67,6 +67,39 @@ func ConnectLXD(url string, args *ConnectionArgs) (InstanceServer, error) {
 	return httpsLXD(url, args)
 }
 
+// ConnectLXDHTTP lets you connect to a VM agent over a VM socket.
+func ConnectLXDHTTP(vsockID int, args *ConnectionArgs, client *http.Client) (InstanceServer, error) {
+	logger.Debugf("Connecting to a VM agent over a VM socket")
+
+	// Use empty args if not specified
+	if args == nil {
+		args = &ConnectionArgs{}
+	}
+
+	// Initialize the client struct
+	server := ProtocolLXD{
+		httpHost:      "https://vm.socket",
+		httpProtocol:  "vsock",
```

Comment:
We should set that:
 - httpHost = "http://custom.socket"
 - httpProtocol = "custom"

Reply:
It should be `https://custom.socket` though as our websocket code checks the prefix.

Reply:
Done